include/stdshims.h

```diff
@@ -7,8 +7,10 @@
 #include <memory>
 #include <utility>
 
-#if (__cplusplus >= 201402L)
+#if (__cplusplus >= 201402L || _MSC_VER)
```

Comment:
VS 2012(`_MSC_VER == 1700`) doesn't have `std::make_unique` while 2013 (`_MSC_VER == 1800`) has. This will cause compile errors when using VS 2012.